frontend/class-frontend.php

```diff
@@ -1042,63 +1042,66 @@ public function adjacent_rel_links() {
 			return;
 		}
 
-		global $wp_query;
-
 		if ( ! is_singular() ) {
-			$url = $this->canonical( false, true, true );
+			$this->rel_links_archive();
+			return;
+		}
+		$this->rel_links_single();
+	}
 
-			if ( is_string( $url ) && $url !== '' ) {
-				$paged = get_query_var( 'paged' );
+	/**
+	 * Output the rel next/prev links for a single post / page.
+	 */
+	private function rel_links_single() {
+		$num_pages = 0;
+		if ( isset( $wp_query->post->post_content ) ) {
+			$num_pages = ( substr_count( $GLOBALS['wp_query']->post->post_content, '<!--nextpage-->' ) + 1 );
+		}
+		if ( $num_pages > 1 ) {
+			$page = intval( get_query_var( 'page' ) );
+			if ( ! $page ) {
+				$page = 1;
+			}
 
-				if ( 0 === $paged ) {
-					$paged = 1;
-				}
+			$url = get_permalink( $GLOBALS['wp_query']->post->ID );
 
-				if ( $paged === 2 ) {
-					$this->adjacent_rel_link( 'prev', $url, ( $paged - 1 ), true );
-				}
+			if ( $page > 1 ) {
+				$this->adjacent_rel_link( 'prev', $url, ( $page - 1 ) );
+			}
+			if ( $page < $num_pages ) {
+				$this->adjacent_rel_link( 'next', $url, ( $page + 1 ) );
+			}
+		}
+	}
 
-				// Make sure to use index.php when needed, done after paged == 2 check so the prev links to homepage will not have index.php erroneously.
-				if ( is_front_page() ) {
-					$url = WPSEO_Sitemaps_Router::get_base_url( '' );
-				}
+	/**
+	 * Output the rel next/prev links for an archive page.
+	 */
+	private function rel_links_archive() {
+		$url = $this->canonical( false, true, true );
 
-				if ( $paged > 2 ) {
-					$this->adjacent_rel_link( 'prev', $url, ( $paged - 1 ), true );
-				}
+		if ( is_string( $url ) && $url !== '' ) {
+			$paged = intval( get_query_var( 'paged' ) );
```

Comment:
Suggestion: use `(int)` instead of `intval()`.
`intval()` is a function and PHP4-style code. `(int)` is a language construct and therefore faster.